pkg/provider/cloud/kubevirt/provider.go

```diff
@@ -96,8 +110,25 @@ func (k *kubevirt) reconcileCluster(cluster *kubermaticv1.Cluster, update provid
 	return cluster, nil
 }
 
-func (k *kubevirt) CleanUpCloudProvider(c *kubermaticv1.Cluster, p provider.ClusterUpdater) (*kubermaticv1.Cluster, error) {
-	return c, nil
+func (k *kubevirt) CleanUpCloudProvider(cluster *kubermaticv1.Cluster, update provider.ClusterUpdater) (*kubermaticv1.Cluster, error) {
+	client, _, err := k.GetClientWithRestConfigForCluster(cluster)
+	if err != nil {
+		return cluster, err
+	}
+
+	if kuberneteshelper.HasFinalizer(cluster, FinalizerNamespace) {
+		if err := deleteNamespace(cluster.Status.NamespaceName, client); err != nil && !kerrors.IsNotFound(err) {
+			return cluster, fmt.Errorf("failed to delete namespace %s: %w", cluster.Status.NamespaceName, err)
+		}
+		cluster, err = update(cluster.Name, func(updatedCluster *kubermaticv1.Cluster) {
+			kuberneteshelper.RemoveFinalizer(updatedCluster, FinalizerNamespace)
+		})
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This can be shortened to `return update(.....)`

Reply:
Done